web/static/js/components/import_export_tray.js

```diff
@@ -0,0 +1,162 @@
+import React from 'react'
+import CopyToClipboard from 'react-copy-to-clipboard'
+
+class ImportExportTray extends React.Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      pageContents: this.props.pageContents,
+      pageSettings: this.props.pageSettings,
+      stringifiedExportData: '',
+      stringifiedImportData: '',
+      exportDataButtonText: 'Copy to Clipboard'
+    }
+
+    this.onExportCopy = this.onExportCopy.bind(this)
+    this.stringifyExportData = this.stringifyExportData.bind(this)
+    this.onImportDataPaste = this.onImportDataPaste.bind(this)
+    this.onImportApply = this.onImportApply.bind(this)
+  }
+
+  componentWillReceiveProps (nextProps) {
+    this.setState({
+      pageContents: nextProps.pageContents,
+      pageSettings: nextProps.pageSettings,
+      stringifiedExportData: '',
+      stringifiedImportData: '',
+      exportDataButtonText: 'Copy to Clipboard'
+    })
+  }
+
+  componentDidMount () {
+    this.stringifyExportData()
+  }
+
+  componentDidUpdate (prevProps, prevState) {
+    if (this.state.stringifiedExportData !== '') return
+
+    this.stringifyExportData()
+  }
```

Comment:
we also rebuild it after receiving new props. basically preparing the data to be copied.